### PR TITLE
[Merged by Bors] - Allow HTTP API to return SSZ blocks

### PR DESCRIPTION
**Pull request by michaelsproul**
## Issue Addressed

Implements https://github.com/ethereum/eth2.0-APIs/pull/125

## Proposed Changes

Optionally return SSZ bytes from the `beacon/blocks` endpoint.

### CODE REVIEW

**Comment by michaelsproul** [comment]
Blocked on upstream PR

**Comment by paulhauner** [comment]
I assume this just adds functionality without removing anything. As such, I don't mind you adding this before the standard API PR is merged. Your call though.

The HTTP API is presently pretty consistent when it comes to having a test for every endpoint and a corresponding "getter" on the [`BeaconNodeHttpClient`](https://github.com/sigp/lighthouse/blob/f819ba54143b02d42316b0b67acee2055bbca44c/common/eth2/src/lib.rs#L82). Up to you if you want to add them.

There's an existing "getter" using SSZ here:

https://github.com/sigp/lighthouse/blob/f819ba54143b02d42316b0b67acee2055bbca44c/common/eth2/src/lib.rs#L859-L878

Test could pretty easily slide in here:

https://github.com/sigp/lighthouse/blob/f819ba54143b02d42316b0b67acee2055bbca44c/beacon_node/http_api/tests/tests.rs#L957-L972

**Comment by michaelsproul** [comment]
Oh yeah! I can add the test and we can merge :sunglasses:

Btw, I've been using this endpoint to grab starting blocks for weak subjectivity sync

**Comment by michaelsproul** [comment]
This is ready for final review and merge!

(Upstream PR looks like it will be merged soon too)

**Review by paulhauner — APPROVED** [review]
LGTM!

**Pull request closed without merging**
